addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_nodes.py

```diff
@@ -243,6 +243,8 @@ def __gather_mesh(blender_object, export_settings):
             blender_mesh = blender_object.to_mesh(bpy.context.scene, True, 'PREVIEW')
         else:
             blender_mesh = blender_object.to_mesh(bpy.context.depsgraph, True)
+        for prop in blender_object.data.keys():
+            blender_mesh[prop] = blender_object.data[prop]
```

Comment:
this would change the state of the scene, right? If so, we should search for another way, maybe make a copy of the mesh and delete it again

Reply:
Related: https://github.com/KhronosGroup/glTF-Blender-IO/pull/413 by @pjoe – in that case the solution was to _avoid_ copying the object.

Reply:
Based on what I remember seeing in testing, blender_mesh ends up being a totally new datablock that is unlinked from the scene a few lines later, which should leave it in the same state. I haven't tested whether to_mesh returns the original datablock if the mesh has no modifiers, but in that case, we'd be assigning properties from the mesh back to itself, which should be a no-op.